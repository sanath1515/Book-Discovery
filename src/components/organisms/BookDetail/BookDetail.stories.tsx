import React from "react";
import BookDetail from "./BookDetail";

export default {
  title: "organisms/BookDetail",
  component: BookDetail,
};

const Template = () => (
  <BookDetail
    image="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/19709E25-7BBE-4E66-8C4B-E83184539737.png"
    authorName="J D Lee"
    fieldName="Chemistry"
    title="Concise Inorganic Chemistry"
    desc="Through a series of recent breakthroughs, deep learning has boosted the entire field of machine learning.Now, even programmers who know close to nothing about this technology can use simple, efficient tools to implement programs capable of learning from data.This practical book shows you how.By using concrete examples, minimal theory, and two production-ready Python frameworks—Scikit-Learn and TensorFlow—author Aurélien Géron helps you gain an intuitive understanding of the concepts and tools for building intelligent systems. You’ll learn a range of techniques, starting with simple linear regression and progressing to deep neural networks. With exercises in each of the concepts and tools for building intelligent systems. You’ll learn a range of techniques, starting with simple linear regression and progressing to deep neural networks. With exercises in each of the concepts and tools for building intelligent systems. You’ll learn a range of techniques, starting with simple linear regression and progressing to deep neural networks. With exercises in each of the concepts and tools for building intelligent systems. You’ll learn a range of techniques, starting with simple linear regression and progressing to deep neural networks."
    releaseDate="12 sep 1998"
    language="English"
    pagesRead={"100"} totalPages={0}  />
);

export const Default = Template.bind({});
