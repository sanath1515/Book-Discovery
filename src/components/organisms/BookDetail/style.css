.round {
  width: 40px;
  height: 40px;
  margin: 5px 28px 6px 30px;
  border-style: solid;
  border-width: 8px;
  border-color: #ebf4f7;
  border-radius: 50%;
  border-image-slice: 1;
  text-align: center;
}
