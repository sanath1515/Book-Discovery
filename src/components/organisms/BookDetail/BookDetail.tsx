import React, { useState } from "react";
import {  CardActions,  CardMedia,  Grid,  makeStyles,  TableCell,  TableRow,  Typography,} from "@material-ui/core";
import { COLORS } from "../../../theme/constants";
import { Rating } from "@material-ui/lab";
import BookmarkIcon from "@material-ui/icons/Bookmark";
import PlaylistAddIcon from "@material-ui/icons/PlaylistAddRounded";
import ReplySharpIcon from "@material-ui/icons/ReplySharp";
import FormatListBulletedRoundedIcon from "@material-ui/icons/FormatListBulletedRounded";
import BookmarkBorderIcon from "@material-ui/icons/BookmarkBorder";
import LanguageIcon from "@material-ui/icons/Language";
import NewReleasesOutlinedIcon from "@material-ui/icons/NewReleasesOutlined";
import theme from "../../../theme/theme";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import ExpandLessIcon from "@material-ui/icons/ExpandLess";
import "./style.css";
import SplitButton from "../../molecules/SplitButton/index";

const useStyles = makeStyles({
  title: {
    width: "auto",
    height: "38px",
    margin: theme.spacing(6, 1, 0, 2),
    color: COLORS.CHARCOAL_GREY,
  },
  image: {
    width: "180px",
    height: "259px",
    margin: theme.spacing(4, 2.5,4,1),
    objectFit: "fill"
  },
  by: {
    width: "15px",
    height: "22px",
    color: COLORS.ICON_500,
  },
  authorName: {
    width: "401px",
    height: "28px",
    color: COLORS.CHARCOAL_GREY,
    fontWeight: "normal",
  },
  field: {
    width: "37px",
    height: "22px",
    color: COLORS.ICON_500,
  },
  fieldName: {
    width: "auto",
    height: "24px",
    color: COLORS.CHARCOAL_GREY,
  },
  desc: {
    width: "154px",
    height: "30px",
    margin: theme.spacing(1, 3, 1, 0),
    color: COLORS.CHARCOAL_GREY,
  },
  descText: {
    width: "727px",
    color: COLORS.CHARCOAL_GREY_100,
    fontWeight: "normal",
  },
  star: {
    width: "20px",
    height: "22px",
    color: COLORS.CHARCOAL_GREY,
  },
  rating: {
    width: "108px",
    height: "22px",
    color: COLORS.CHARCOAL_GREY,
  },
  release: {
    fontSize: "15px",
    color: COLORS.CHARCOAL_GREY_100,
    paddingBottom:theme.spacing(1)
  },
  releaseDate: {
    width: "169px",
    height: "22px",
    marginLeft: theme.spacing(0),
    color: COLORS.CHARCOAL_GREY_100,
  },
  languageIcon: {
    fontSize: "15px",
    color: COLORS.CHARCOAL_GREY_100,
    paddingBottom:theme.spacing(1)
  },
  language: {
    color: COLORS.CHARCOAL_GREY_100,
    width: "46px",
    height: "22px",
  },
  reply: {
    color: COLORS.ICON_500,
    paddingRight: theme.spacing(3),
    fontSize: "20px",
  },
  playlist: {
    color: COLORS.ICON_500,
    paddingRight: theme.spacing(3),
    fontSize: "20px",
  },
  bookmark: {
    color: COLORS.PRIMARY_COLOR_500,
    paddingRight: theme.spacing(3),
    fontSize: "20px",
    cursor: "pointer",
  },
  bookmarkBorder: {
    color: COLORS.ICON_500,
    paddingRight: theme.spacing(3),
    fontSize: "20px",
    cursor: "pointer",
  },
  list: {
    color: COLORS.ICON_500,
    fontSize: "18px",
  },
  actions: {
    padding: theme.spacing(1.5),
  },
  firstActions: {
    marginTop: theme.spacing(2),
    paddingBottom: theme.spacing(0.5),
  },
  grid: {
    display: "flex",
  },
  flex: {
    display: "flex",
    height: "50px",
  },
  text: {
    width: "68px",
    height: "24px",
    margin: theme.spacing(0, 0, 0, 65),
    color: COLORS.PRIMARY_COLOR_500,
    cursor: "pointer"
  },
  icon: {
    color: COLORS.PRIMARY_COLOR_500,
    fontSize: "20px",
    cursor: "pointer"
  },
  pages: {
    width: "103px",
    height: "18px",
    color: COLORS.CHARCOAL_GREY_100,
    textTransform: "none",
    margin: theme.spacing(2, 1.5),
  },
  spacing: {
    margin: theme.spacing(-1),
    padding: theme.spacing(1, 0.8),
  },
  descTextEnd: {
    color: COLORS.CHARCOAL_GREY_100,
    fontWeight: "normal",
  },
  border: {
    border: "none",
    marginLeft: "auto"
  },
  info:{
    width:"100%"
  }
});

export type BookDetailProps = {
  desc: string;
  title: string;
  authorName: string;
  fieldName: string;
  releaseDate: string;
  language: string;
  image: string;
  pagesRead: string;
  totalPages: number;
  className?: string;
};
const BookDetail: React.FC<BookDetailProps> = ({
  desc,
  title,
  authorName,
  fieldName,
  releaseDate,
  language,
  image,
  pagesRead,
  totalPages
}) => {
  const desc1 = desc.slice(0, 294);
  const desc2 = desc.slice(294, 638);
  const desc3 = desc.slice(638, 847);
  const [bookMark, setBookMark] = useState(false);
  const [isSeeMore, setIsSeeMore] = useState(false);

  const onBookmarkClick = () => {
    setBookMark(!bookMark);
  };
  const handleOnClick = () => {
    setIsSeeMore(!isSeeMore);
  };
  const classes = useStyles();
  return (
    <React.Fragment>
      <Grid className={classes.grid}>
        <CardMedia className={classes.image} component="img" image={image} />
        <div className={classes.info}>
          <div className={classes.flex}>
            <Typography variant="h3" className={classes.title}>
              {title}
            </Typography>
            <TableCell align="right" className={classes.border}>
              <TableRow>
                <div className="round"></div>
              </TableRow>
              <TableRow>
                <Typography variant="overline" className={classes.pages}>
                  {pagesRead} read / {totalPages} pages
                </Typography>
              </TableRow>
            </TableCell>
          </div>
          <CardActions className={classes.firstActions}>
            <Typography variant="body2" className={classes.by}>
              by
            </Typography>
            <Typography className={classes.authorName} variant="h6">
              {authorName}
            </Typography>
          </CardActions>
          <CardActions className={classes.actions}>
            <Typography variant="body2" className={classes.field}>
              Field:
            </Typography>
            <Typography className={classes.fieldName} variant="subtitle2">
              {fieldName}
            </Typography>
          </CardActions>
          <CardActions className={classes.actions}>
            <Rating
              name="half-rating-read"
              defaultValue={4.5}
              precision={0.5}
              size="small"
              readOnly
            />
            <Typography className={classes.star} variant="body1">
              4.5
            </Typography>
            <Typography className={classes.rating} variant="body2">
              ( 1,204 reviews )
            </Typography>
          </CardActions>
          <CardActions className={classes.actions}>
            <NewReleasesOutlinedIcon className={classes.release} />
            <Typography variant="body2" className={classes.releaseDate}>
              Release Date: {releaseDate}
            </Typography>
            <LanguageIcon className={classes.languageIcon} />
            <Typography variant="body2" className={classes.language}>
              {language}
            </Typography>
          </CardActions>
          <CardActions className={classes.actions}>
            {bookMark && (
              <BookmarkIcon
                className={classes.bookmark}
                onClick={onBookmarkClick}
                data-testid="bookmark"
              />
            )}
            {!bookMark && (
              <BookmarkBorderIcon
                className={classes.bookmarkBorder}
                onClick={onBookmarkClick}
                data-testid="bookmark"
              />
            )}
            <PlaylistAddIcon className={classes.playlist} />
            <ReplySharpIcon className={classes.reply} />
            <FormatListBulletedRoundedIcon className={classes.list} />
          </CardActions>
          <CardActions className={classes.actions}>
            <SplitButton items={["Start Reading", "Finish Reading"]} />
          </CardActions>
        </div>
      </Grid>
      <Typography variant="h4" className={classes.desc}>
        Book Description
      </Typography>
      <Typography variant="h6" className={classes.descText}>
        {desc1}
      </Typography>
      <Typography variant="h6" className={classes.descText}>
        {desc2}
      </Typography>
      {!isSeeMore && (
        <CardActions className={classes.spacing}>
          <Typography variant="h6" className={classes.descTextEnd}>
            {desc3.slice(0, desc3.length - 175)}
          </Typography>
          <Typography
            variant="subtitle2"
            className={classes.text}
            onClick={handleOnClick}
            data-testid="seemore"
          >
            See more
          </Typography>
          <ExpandMoreIcon className={classes.icon} onClick={handleOnClick} />
        </CardActions>
      )}
      {isSeeMore && (
        <React.Fragment>
          <Typography variant="h6" className={classes.descTextEnd}>
            {desc3.slice(0, desc3.length-25)}
          </Typography>
          <CardActions className={classes.spacing}>
            <Typography variant="h6" className={classes.descTextEnd}>
              {desc3.slice(desc3.length - 25,desc.length)}
            </Typography>
            <Typography
              variant="subtitle2"
              className={classes.text}
              onClick={handleOnClick}
             
            >
              See less
            </Typography>
            <ExpandLessIcon className={classes.icon} onClick={handleOnClick} />
          </CardActions>
        </React.Fragment>
      )}
    </React.Fragment>
  );
};

export default BookDetail;
