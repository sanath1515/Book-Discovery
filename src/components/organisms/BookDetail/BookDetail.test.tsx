import { fireEvent, render, screen } from "@testing-library/react";
import BookDetail from "./BookDetail";

describe("Book Detail Testing", () => {
  it("Matches Snapshot", () => {
    const { container } = render(
      <BookDetail
        image="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/19709E25-7BBE-4E66-8C4B-E83184539737.png"
        authorName="J D Lee"
        fieldName="Chemistry"
        title="Concise Inorganic Chemistry"
        desc=""
        releaseDate="12 sep 1998"
        language="English"
        pagesRead={""} totalPages={200}      />
    );
    expect(container).toMatchSnapshot();
  });

  it("Component to be in the Document", () => {
    render(
      <BookDetail
        image="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/19709E25-7BBE-4E66-8C4B-E83184539737.png"
        authorName="J D Lee"
        fieldName="Chemistry"
        title="Concise Inorganic Chemistry"
        desc=""
        releaseDate="12 sep 1998"
        language="English"
        pagesRead={""} totalPages={0}      />
    );
    expect(screen.getByText("Concise Inorganic Chemistry")).toBeInTheDocument();
  });

  it("Should click bookmark correctly", () => {
    const { getByTestId } = render(
      <BookDetail
        image="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/19709E25-7BBE-4E66-8C4B-E83184539737.png"
        authorName="J D Lee"
        fieldName="Chemistry"
        title="Concise Inorganic Chemistry"
        desc="Through a series of recent breakthroughs, deep learning has boosted the entire field of machine learning.Now, even programmers who know close to nothing about this technology can use simple, efficient tools to implement programs capable of learning from data.This practical book shows you how.By using concrete examples, minimal theory, and two production-ready Python frameworks—Scikit-Learn and TensorFlow—author Aurélien Géron helps you gain an intuitive understanding of the concepts and tools for building intelligent systems. You’ll learn a range of techniques, starting with simple linear regression and progressing to deep neural networks. With exercises in each of the concepts and tools for building intelligent systems. You’ll learn a range of techniques, starting with simple linear regression and progressing to deep neural networks. With exercises in each of the concepts and tools for building intelligent systems. You’ll learn a range of techniques, starting with simple linear regression and progressing to deep neural networks. With exercises in each of the concepts and tools for building intelligent systems. You’ll learn a range of techniques, starting with simple linear regression and progressing to deep neural networks."
        releaseDate="12 sep 1998"
        language="English"
        pagesRead={"100"} totalPages={0}      />
    );
    expect(fireEvent.click(getByTestId("bookmark"))).toBeTruthy();
  });
  it("Should click seemore correctly", () => {
    const { getByTestId } = render(
      <BookDetail
        image="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/19709E25-7BBE-4E66-8C4B-E83184539737.png"
        authorName="J D Lee"
        fieldName="Chemistry"
        title="Concise Inorganic Chemistry"
        desc="Through a series of recent breakthroughs, deep learning has boosted the entire field of machine learning.Now, even programmers who know close to nothing about this technology can use simple, efficient tools to implement programs capable of learning from data.This practical book shows you how.By using concrete examples, minimal theory, and two production-ready Python frameworks—Scikit-Learn and TensorFlow—author Aurélien Géron helps you gain an intuitive understanding of the concepts and tools for building intelligent systems. You’ll learn a range of techniques, starting with simple linear regression and progressing to deep neural networks. With exercises in each of the concepts and tools for building intelligent systems. You’ll learn a range of techniques, starting with simple linear regression and progressing to deep neural networks. With exercises in each of the concepts and tools for building intelligent systems. You’ll learn a range of techniques, starting with simple linear regression and progressing to deep neural networks. With exercises in each of the concepts and tools for building intelligent systems. You’ll learn a range of techniques, starting with simple linear regression and progressing to deep neural networks."
        releaseDate="12 sep 1998"
        language="English"
        pagesRead={"100"} totalPages={0}      />
    );
    expect(fireEvent.click(getByTestId("seemore"))).toBeTruthy();
  });
});
