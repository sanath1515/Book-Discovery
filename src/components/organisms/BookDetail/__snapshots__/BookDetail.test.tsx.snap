// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Detail Testing Matches Snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-grid-22"
  >
    <img
      class="MuiCardMedia-root makeStyles-image-2 MuiCardMedia-media MuiCardMedia-img"
      src="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/19709E25-7BBE-4E66-8C4B-E83184539737.png"
    />
    <div
      class="makeStyles-info-30"
    >
      <div
        class="makeStyles-flex-23"
      >
        <h3
          class="MuiTypography-root makeStyles-title-1 MuiTypography-h3"
        >
          Concise Inorganic Chemistry
        </h3>
        <td
          class="MuiTableCell-root makeStyles-border-29 MuiTableCell-alignRight"
        >
          <tr
            class="MuiTableRow-root"
          >
            <div
              class="round"
            />
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <span
              class="MuiTypography-root makeStyles-pages-26 MuiTypography-overline"
            >
              
               read / 
              200
               pages
            </span>
          </tr>
        </td>
      </div>
      <div
        class="MuiCardActions-root makeStyles-firstActions-21 MuiCardActions-spacing"
      >
        <p
          class="MuiTypography-root makeStyles-by-3 MuiTypography-body2"
        >
          by
        </p>
        <h6
          class="MuiTypography-root makeStyles-authorName-4 MuiTypography-h6"
        >
          J D Lee
        </h6>
      </div>
      <div
        class="MuiCardActions-root makeStyles-actions-20 MuiCardActions-spacing"
      >
        <p
          class="MuiTypography-root makeStyles-field-5 MuiTypography-body2"
        >
          Field:
        </p>
        <h6
          class="MuiTypography-root makeStyles-fieldName-6 MuiTypography-subtitle2"
        >
          Chemistry
        </h6>
      </div>
      <div
        class="MuiCardActions-root makeStyles-actions-20 MuiCardActions-spacing"
      >
        <span
          aria-label="4.5 Stars"
          class="MuiRating-root MuiRating-sizeSmall MuiRating-readOnly"
          role="img"
        >
          <span
            class="MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal"
          >
            <span
              style="width: 50%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </span>
        <p
          class="MuiTypography-root makeStyles-star-9 MuiTypography-body1"
        >
          4.5
        </p>
        <p
          class="MuiTypography-root makeStyles-rating-10 MuiTypography-body2"
        >
          ( 1,204 reviews )
        </p>
      </div>
      <div
        class="MuiCardActions-root makeStyles-actions-20 MuiCardActions-spacing"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-release-11"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M23 12l-2.44-2.78.34-3.68-3.61-.82-1.89-3.18L12 3 8.6 1.54 6.71 4.72l-3.61.81.34 3.68L1 12l2.44 2.78-.34 3.69 3.61.82 1.89 3.18L12 21l3.4 1.46 1.89-3.18 3.61-.82-.34-3.68L23 12zm-4.51 2.11l.26 2.79-2.74.62-1.43 2.41L12 18.82l-2.58 1.11-1.43-2.41-2.74-.62.26-2.8L3.66 12l1.85-2.12-.26-2.78 2.74-.61 1.43-2.41L12 5.18l2.58-1.11 1.43 2.41 2.74.62-.26 2.79L20.34 12l-1.85 2.11zM11 15h2v2h-2zm0-8h2v6h-2z"
          />
        </svg>
        <p
          class="MuiTypography-root makeStyles-releaseDate-12 MuiTypography-body2"
        >
          Release Date: 
          12 sep 1998
        </p>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-languageIcon-13"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
          />
        </svg>
        <p
          class="MuiTypography-root makeStyles-language-14 MuiTypography-body2"
        >
          English
        </p>
      </div>
      <div
        class="MuiCardActions-root makeStyles-actions-20 MuiCardActions-spacing"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-bookmarkBorder-18"
          data-testid="bookmark"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-playlist-16"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 10H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1zm5 8v-3c0-.55-.45-1-1-1s-1 .45-1 1v3h-3c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3zM3 16h6c.55 0 1-.45 1-1s-.45-1-1-1H3c-.55 0-1 .45-1 1s.45 1 1 1z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-reply-15"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 9V5l-7 7 7 7v-4.1c5 0 8.5 1.6 11 5.1-1-5-4-10-11-11z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-list-19"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M4 10.5c-.83 0-1.5.67-1.5 1.5s.67 1.5 1.5 1.5 1.5-.67 1.5-1.5-.67-1.5-1.5-1.5zm0-6c-.83 0-1.5.67-1.5 1.5S3.17 7.5 4 7.5 5.5 6.83 5.5 6 4.83 4.5 4 4.5zm0 12c-.83 0-1.5.68-1.5 1.5s.68 1.5 1.5 1.5 1.5-.68 1.5-1.5-.67-1.5-1.5-1.5zM8 19h12c.55 0 1-.45 1-1s-.45-1-1-1H8c-.55 0-1 .45-1 1s.45 1 1 1zm0-6h12c.55 0 1-.45 1-1s-.45-1-1-1H8c-.55 0-1 .45-1 1s.45 1 1 1zM7 6c0 .55.45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1H8c-.55 0-1 .45-1 1z"
          />
        </svg>
      </div>
      <div
        class="MuiCardActions-root makeStyles-actions-20 MuiCardActions-spacing"
      >
        <div
          class="MuiGrid-root makeStyles-grid-35 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              aria-label="split button"
              class="MuiButtonGroup-root makeStyles-btn-31 MuiButtonGroup-contained"
              role="group"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained makeStyles-btn1-32"
                data-testid="btn-click"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Start Reading
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained makeStyles-btn-31 MuiButton-containedSizeSmall MuiButton-sizeSmall"
                data-testid="openDropdown"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <h4
    class="MuiTypography-root makeStyles-desc-7 MuiTypography-h4"
  >
    Book Description
  </h4>
  <h6
    class="MuiTypography-root makeStyles-descText-8 MuiTypography-h6"
  />
  <h6
    class="MuiTypography-root makeStyles-descText-8 MuiTypography-h6"
  />
  <div
    class="MuiCardActions-root makeStyles-spacing-27 MuiCardActions-spacing"
  >
    <h6
      class="MuiTypography-root makeStyles-descTextEnd-28 MuiTypography-h6"
    />
    <h6
      class="MuiTypography-root makeStyles-text-24 MuiTypography-subtitle2"
      data-testid="seemore"
    >
      See more
    </h6>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root makeStyles-icon-25"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
      />
    </svg>
  </div>
</div>
`;
